sup3r/preprocessing/exogenous_data_handling.py

```diff
@@ -0,0 +1,83 @@
+"""Sup3r exogenous data handling"""
+
+import logging
+import numpy as np
+
+from sup3r.utilities.topo import TopoExtract
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExogenousDataHandler:
+    """Class to extract exogenous features for multistep forward passes. e.g.
+    Multiple topography arrays at different resolutions for multiple spatial
+    enhancement steps."""
+
+    def __init__(self, file_paths, features, source_h5, s_enhancements,
+                 agg_factors, target=None, shape=None, raster_file=None,
+                 max_delta=20, input_handler=None):
+        """
+        Parameters
+        ----------
+        file_paths : str | list
+            A single source h5 wind file to extract raster data from or a list
```

Comment:
This isn't always a wind file, and actually you might want a clause saying "this is typically low-res WRF output or GCM netcdf data files that is source low-resolution data intended to be sup3r resolved"... Just to further differentiate it from source_h5. 

And actually I realize now that I wrote this docstring originally for the TopoExtract utility so if you could update it there too that would be great :)